src/org/thoughtcrime/securesms/NicknameHandler.java

```diff
@@ -0,0 +1,70 @@
+package org.thoughtcrime.securesms;
+
+import android.content.Context;
+
+import org.thoughtcrime.securesms.database.DatabaseFactory;
+import org.thoughtcrime.securesms.database.RecipientDatabase;
+import org.thoughtcrime.securesms.recipients.Recipient;
+
+
+public class NicknameHandler {
```

Comment:
I don't understand the point of this whole class. It seems to be some kind of database helper, but right now it does almost nothing.

Reply:
this method allows us to write tests and check if nickname feature works, if we dont do this, fragment will right away call databases and we can not test fragments

Reply:
But couldn't the "setDatabase" method be on the fragment?